.github/workflows/robot/internal/bot/backport.go

```diff
@@ -0,0 +1,173 @@
+/*
+Copyright 2022 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package bot
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"log"
+	"net/url"
+	"os/exec"
+	"path"
+	"strings"
+	"text/template"
+
+	"github.com/gravitational/trace"
+)
+
+func (b *Bot) Backport(ctx context.Context) error {
+	pull, err := b.c.GitHub.GetPullRequest(ctx,
+		b.c.Environment.Organization,
+		b.c.Environment.Repository,
+		b.c.Environment.Number)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	branches := findBranches(pull.UnsafeLabels)
+
+	var statuses []Status
+	for _, branch := range branches {
+		status := "Success"
+		number, err := b.backportBranch(ctx,
+			b.c.Environment.Organization,
+			b.c.Environment.Repository,
+			b.c.Environment.Number,
+			branch)
+		if err != nil {
+			status = "Failure"
+			log.Printf("Failed to backport %v to %v: %v.", b.c.Environment.Number, branch, err)
+		}
+
+		statuses = append(statuses, Status{
+			Status: status,
+			Branch: branch,
+			Link: url.URL{
+				Scheme: "https",
+				Host:   "github.com",
+				Path:   path.Join("gravitational", "teleport", "pull", number),
+			},
+		})
+	}
+
+	err = b.updatePullRequest(ctx,
+		b.c.Environment.Organization,
+		b.c.Environment.Repository,
+		b.c.Environment.Number,
+		statuses)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	return nil
+}
+
+func findBranches(labels []string) []string {
+	var branches []string
+
+	for _, label := range labels {
+		if !strings.HasPrefix(label, "backport/") {
+			continue
+		}
+
+		branches = append(branches, strings.TrimPrefix(label, "backport/"))
+	}
+
+	return branches
+}
+
+func (b *Bot) backportBranch(ctx context.Context, organization string, repository string, number int, branch string) (string, error) {
+	head := fmt.Sprintf("bot/backport-%v", number)
+
+	// Create backport branch and cherry-pick commits onto it.
+	// TODO(russjones): Fix backport logic here.
+	if err := git("checkout", "-b", head); err != nil {
+		return "", trace.Wrap(err)
+	}
+	commits, err := b.c.GitHub.ListCommits(ctx,
+		organization,
+		repository,
+		number)
+	if err != nil {
+		return "", trace.Wrap(err)
+	}
+	for _, commit := range commits {
+		if err := git("cherry-pick", commit); err != nil {
+			return "", trace.Wrap(err)
+		}
+	}
+	if err := git("push", "origin", head); err != nil {
```

Comment:
Don't you need a -u here?

Reply:
Hrm, I don't think so. I've never actually had to use the `-u` flag?

I'll double check when I do some manual testing.